common/djangoapps/course_modes/models.py

```diff
@@ -768,6 +768,10 @@ def is_eligible_for_certificate(cls, mode_slug):
         """
         ineligible_modes = [cls.AUDIT]
 
+        from lms.djangoapps.certificates.models import CertificateStatuses
+        if mode_slug == cls.HONOR and status == CertificateStatuses.downloadable:
```

Comment:
Move this check over `ineligible_modes.append(cls.HONOR)`, don't append `Honor` mode in ineligible modes list if we are trying to regenerate honor certificate.